### PR TITLE
feat: Support `OutOfOrderTimeWindow` field for PrometheusAgent

### LINKED ISSUE #6840: Add OutOfOrderTimeWindow field support for PrometheusAgent
### Component(s)

PrometheusAgent

### What is missing? Please describe.

Add `OutOfOrderTimeWindow` field support for PrometheusAgent. This is added in Prometheus 2.54

details: https://github.com/prometheus/prometheus/pull/14094

### Describe alternatives you've considered.

_No response_

## Environment
Kubernetes Version:
Prometheus-Operator Version:

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Support `OutOfOrderTimeWindow` field for PrometheusAgent

Closes #6840 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Support `OutOfOrderTimeWindow` field for PrometheusAgent
```

### CODE REVIEW

File: pkg/apis/monitoring/v1alpha1/prometheusagent_types.go [hunk 1]
```diff
@@ -103,4 +103,10 @@ type PrometheusAgentSpec struct {
 	Mode *string `json:"mode,omitempty"`
 
 	monitoringv1.CommonPrometheusFields `json:",inline"`
+
+	// Defines the runtime reloadable configuration of the timeseries database(TSDB).
+	// It requires PrometheusAgent >= v2.54.0.
```
Comment (slashpai) [inline pkg/apis/monitoring/v1alpha1/prometheusagent_types.go:108]:
```suggestion
	// It requires Prometheus >= v2.54.0.
```

**Review by slashpai — COMMENTED** [review]
lgtm, just a nit
@ArthurSens  can you take a look too?

**Review by simonpasquier — COMMENTED** [review]
Hmm shouldn't we remove the `TSDB` field from `PrometheusSpec` too?

**Comment by dongjiang1989** [comment]
move TSDB to common？

**Comment by ArthurSens** [comment]
> move TSDB to common？

That makes sense to me :)

**Comment by dongjiang1989** [comment]
Fixed it done. Move `TSDBSpec` to `CommonPrometheusFields`
@simonpasquier @ArthurSens  Please re-check it, thanks.

**Review by simonpasquier — APPROVED** [review]

**Pull request merged**